FusionIIIT/applications/otheracademic/models.py

```diff
@@ -35,11 +35,109 @@ class LeaveFormTable(models.Model):
     leave_type = models.CharField(max_length=20, choices=LEAVE_TYPES)
     approved = models.BooleanField()
     rejected = models.BooleanField()
+    hod= models.CharField(max_length=100)
 
     class Meta:
         db_table='LeaveFormTable'
         
 
+class LeavePG(models.Model):
+    """
+    Records information related to student leave requests.
+
+    'leave_from' and 'leave_to' store the start and end date of the leave request.
+    'date_of_application' stores the date when the leave request was applied.
+    'related_document' stores any related documents or notes for the leave request.
+    'place' stores the location where the leave is requested.
+    'reason' stores the reason for the leave request.
+    'leave_type' stores the type of leave from a dropdown.
+    """
+    LEAVE_TYPES = (
+        ('Casual', 'Casual'),
+        ('Medical', 'Medical'),
+        ('Vacation', 'Vacation'),
+        ('Duty', 'Duty')
+        
+    )
+    
+
+
+    student_name = models.CharField(max_length=100)
+    roll_no = models.ForeignKey(ExtraInfo, on_delete=models.CASCADE)
+    programme = models.CharField(max_length=100)
+    discipline = models.CharField(max_length=100)
+    Semester = models.CharField(max_length=100)
+    date_from = models.DateField()
+    date_to = models.DateField()
+    date_of_application = models.DateField()
+    upload_file = models.FileField(upload_to='leave_doc')
+    address = models.CharField(max_length=100)
+    purpose = models.TextField()
+    leave_type = models.CharField(max_length=20, choices=LEAVE_TYPES)
+    ta_supervisor = models.CharField(max_length=100)
+    mobile_no = models.CharField(max_length=100)
+    parent_mobile_no = models.CharField(max_length=100)
+    alt_mobile_no = models.CharField(max_length=100)
+    ta_approved = models.BooleanField()
+    ta_rejected = models.BooleanField()
+    hod_approved = models.BooleanField()
```

Comment:
please provide comments about the remaning fields as well.

Reply:
The above fields are used according to leave form of mtech of our college where:-
alt_mobile_no is required ,
ta_approved/ta_rejected:- In procedure of mtech leave when leave form is applied first it goes to respective TA,then it goes to HOD for approval.
So, i used above field for track of event call  by them.